ambry-cloud/src/main/java/com/github/ambry/cloud/CloudBlobStoreV2.java

```diff
@@ -63,10 +79,145 @@ public CloudBlobStoreV2(VerifiableProperties properties, MetricRegistry metricRe
     this.blobContainerClient = blobContainerClient;
   }
 
+  public static final String ACCOUNT_ID = "accountId";
+  public static final String CONTAINER_ID = "containerId";
+  public static final String CRC = "crc";
+  public static final String EXPIRATION_TIME = "expirationTimeInMs";
+  public static final String LIFE_VERSION = "lifeVersion";
+  public static final String OPERATION_TIME = "operationTimeMs";
+  public static final String SIZE = "size";
+
+  /**
+   * @return a {@link HashMap} of metadata key-value pairs.
+   */
+  public Map<String, String> messageInfoToMap(MessageInfo messageInfo) {
+    HashMap<String, String> hashMap = new HashMap<>();
+    hashMap.put(ACCOUNT_ID, String.valueOf(messageInfo.getAccountId()));
+    hashMap.put(CONTAINER_ID, String.valueOf(messageInfo.getContainerId()));
+    hashMap.put(CRC, String.valueOf(messageInfo.getCrc()));
+    hashMap.put(EXPIRATION_TIME, String.valueOf(messageInfo.getExpirationTimeInMs()));
+    hashMap.put(LIFE_VERSION, String.valueOf(messageInfo.getLifeVersion()));
+    hashMap.put(OPERATION_TIME, String.valueOf(messageInfo.getOperationTimeMs()));
+    hashMap.put(SIZE, String.valueOf(messageInfo.getSize()));
+    return hashMap;
+  }
+
+  /**
+   * Detects duplicates in list of {@link MessageInfo}
+   * @param messageInfos list of {@link MessageInfo} to check for duplicates
+   */
+  protected void checkDuplicates(List<MessageInfo> messageInfos) {
+    if (messageInfos.size() < 2) {
+      return;
+    }
+    HashMap<String, MessageInfo> seenKeys = new HashMap<>();
```

Comment:
nit: `HashSet` may be sufficient

Reply:
Actually, I wish to print out the full message info along with key and that's why its a map